_posts/2020/11/2020-11-23-python-repository.md

```diff
@@ -0,0 +1,140 @@
+---
+layout: post
+date: 2020-11-23
+title: "Some name"
+author: olenagerasimova
+description: This blog post shows how binary and pypi repositories can be used.
+keywords:
+  - artipie
+  - private
+  - pypi
+  - python
+  - repository
+  - registry
+  - binary
+---
+
+Being a student I dreamed about something like Artipie, I needed some private centralized storage 
+where I could have kept both binary or text data files and Python packages. We were conducting a research
+and used to work with a lot of data from experimental measurements and used Python to process and 
+visualise them. My university colleagues from that time were mathematicians, they did not have strong
+development background and were absolutely fine with passing data and code via flash drive or email.
+All my efforts to inculcate VCS (GitHub, for example, provides free Pro accounts for students 
+and professors) and some cloud storage usage were always postponed to the brighter future. I've 
+quit the postgraduate program and never saw that future to come.
+
+Now, I'd like to show how I could have used Artipie being that student. In the example I'll use 
+[Artipie Central](http://central.artipie.com/), but the same will work with an Artipie instance 
+running on your private server in the private network.
+
+## Python Repository
+
+Artipie supports [PyPi repository](https://github.com/artipie/artipie/tree/master/examples/pypi) which
+`twine` and `pip` can perfectly understand. This means that you can work with Artipie Python 
+repository exactly as you do while installing or publishing packages from or to [PyPI](https://pypi.org/) 
+and [TestPyPI](https://test.pypi.org/) repos. Here is my PyPi repository configuration from 
+Artipie Central:
+
+```yaml
+repo:
+  type: pypi
+  storage: default
+  permissions:
+    olenagerasimova:
+      - download
+      - upload
+    "*":
+      - download
+```
+
+Upload is available for me only, but anyone can download any package, more details about permission 
+configuration can be found [here](https://github.com/artipie/artipie#repository-permissions). 
+
+To make sure this repository exists and works, open [index page](https://central.artipie.com/olenagerasimova/pypi) 
+from your browser, packages list should be displayed there.
+
+I've created a small Python example project, source code is available [here](https://github.com/artipie/pypi-example), 
+the main idea of the example is from my university times: download three data files from Artipie Central 
+(the next part is about how these files got there), read the numbers into arrays and 
+use these arrays to draw a plot. Let's use `pip` to install this example package and run it:
+
+```commandline
+python3 -m pip install --index-url https://central.artipie.com/olenagerasimova/pypi/ pypiexample
+python3 -m pypiexample
+``` 
+
+I hope that you see a polar plot with three curves, visualisation of our data files. 
```

Comment:
@olenagerasimova I'd suggest to rephrase it to "If you run these commands then you will see...", something like that. I was just reading the text and didn't see the plot!